### PR TITLE
#169649045 -User should be able to Chat with a supplier and line manager

### PR DESCRIPTION
**Pull request by sabin18**
#### What does this PR do?
PR to allow a user to Chat with a supplier and line manager
#### Description of Task to be completed?
- add a private chat function
#### How should this be manually tested?
- clone the repository: git clone https://github.com/andela/caret-bn-backend.git
- checkout to branch ft-get-stats-trips-168781712
- run sequelize db:migrate
- run sequelize db:seed:all
- use this route to chat `http://localhost:3000/`
- use this route to get private chat `{{host}}/api/v1/chats/private`
- add token  in your local storage of browser
- add in your local storage of your browser a userId of the person you want to chat with.
#### Any background context you want to provide?
`N/A`
#### What are the relevant pivotal tracker stories?
[#169649045](https://www.pivotaltracker.com/story/show/169649045)
#### Screenshots (if appropriate)
Screen Shot 2019-11-13 at 17 16 21
Screen Shot 2019-11-14 at 09 35 49

#### Questions:
`N/A`

### CODE REVIEW

File: src/database/seeders/20191114085254-chat.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      {
+        userId: 4,
+        receiverId: 3,
+        createdAt: new Date(),
+        updatedAt: new Date()
+      }
+  ])
+  ]),
+  down: (queryInterface, Sequelize) => Promise.all([
+    queryInterface.bulkDelete('bookmarks', null, {})
```
Comment (nignanthomas) [inline src/database/seeders/20191114085254-chat.js]: It should be `'chats'` not `'bookmarks'`.

File: src/index.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ io.on('connect', socket => {
   });
 });
 
+
```
Comment (nignanthomas) [inline src/index.js]: Remove this line.

File: src/database/migrations/20191106111952-create-chats.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ module.exports = {
       userId: {
         type: Sequelize.INTEGER
       },
+      receiverId: {
+        type: Sequelize.INTEGER
+      },
```
Comment (nignanthomas) [inline src/database/migrations/20191106111952-create-chats.js]: Please create a new migration file to add this column. Thanks.

**Review by nignanthomas — CHANGES_REQUESTED** [review]
That is some really nice stuff here @sabin18 
I left some comments. Kindly work on that.
And try your best to raise the coverage

**Review by nignanthomas — APPROVED** [review]

**Review by ngireric123 — APPROVED** [review]
@sabin18 you did a great job but try to increase the coverage

**Review by alainmateso — APPROVED** [review]

**Pull request merged**